### PR TITLE
feat: add huaweicloud metadata support.

### PR DESCRIPTION
**Pull request by Colstuwjx**
Signed-off-by: colstuwjx <colstuwjx@gmail.com>

- Enhancement

## What does this PR do?

This PR is going to add Huawei cloud metadata provider support.

## Why is it important?

There are lots of company are using huaweicloud ecs, and this PR will help them collect ECS level metadata.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Review by jsoriano — COMMENTED** [review]
Hey @Colstuwjx,

Thanks for your contribution! Code-wise it looks good to me, but could you please add a changelog entry in `CHANGELOG.next.asciidoc`?

@kaiyan-sheng could you please also have a second look?

**Comment by kaiyan-sheng** [comment]
Code looks good to me too. One question, @jsoriano should Huawei be enabled by default? I remember we have to disable Alibaba Cloud and Tencent cloud providers because they require to access a remote host.

**Comment by jsoriano** [comment]
> should Huawei be enabled by default? I remember we have to disable Alibaba Cloud and Tencent cloud providers because they require to access a remote host.

This provider gets the data from 169.254.169.254, as for AWS, Azure or GCP, so I guess that it can be enabled by default.

File: libbeat/processors/add_cloud_metadata/provider_huawei_cloud.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		huaweiCloudMetadataJSONURI := "/openstack/latest/meta_data.json"
+
+		huaweiCloudSchema := func(m map[string]interface{}) common.MapStr {
+			m["service"] = common.MapStr{
+				"name": "ECS",
+			}
+			return common.MapStr{"cloud": m}
+		}
+
+		urls, err := getMetadataURLs(c, huaweiCloudMetadataHost, []string{
```
Comment (jsoriano) [inline libbeat/processors/add_cloud_metadata/provider_huawei_cloud.go:58]: Other providers that get the metadata from `169.254.169.254` use `metadataHost` global variable. But it is ok for me to have this defined here too.
```suggestion
		urls, err := getMetadataURLs(c, metadataHost, []string{
```

**Review by jsoriano — APPROVED** [review]

**Comment by jsoriano** [comment]
> could you please add a changelog entry in `CHANGELOG.next.asciidoc`?

I have added it, this would be ready to merge.

**Comment by Colstuwjx** [comment]
> Code looks good to me too. One question, @jsoriano should Huawei be enabled by default? I remember we have to disable Alibaba Cloud and Tencent cloud providers because they require to access a remote host.

HuaweiCloud is similar to openstack, it's using 169.254.169.254, which should be local call, so I think it's ok to enable by default.

I've been rebased with master! Now I think it's ready to merge!

**Comment by Colstuwjx** [comment]
Fixed a rebase issue, the PR commits are clear right now.

**Review by jsoriano — APPROVED** [review]

**Comment by Colstuwjx** [comment]
@jsoriano @kaiyan-sheng Anything updates? If there is any issue left, please let me know.

**Review by kaiyan-sheng — APPROVED** [review]
Looks good to me.

**Pull request merged**